pokete.py

```diff
@@ -985,6 +985,88 @@ def __call__(self):
         Station.choosen.unchoose()
 
 
+class Dex:
+    def __init__(self, map):
+        self.box = ChooseBox(map.height-3, 35, "Poketedex")
+        self.detail_box = Box(16, 35)
+        self.map = map
+        self.idx = 0
+        self.obs = []
+        self.detail_info = se.Text("", state="float")
+        self.detail_desc = se.Text("", state="float")
+        self.detail_box.add_ob(self.detail_info, 16, 1)
+        self.detail_box.add_ob(self.detail_desc, 3, 7)
+
+    def add_c_obs(self):
+        self.box.add_c_obs(self.obs[self.idx*(self.box.height-2):(self.idx+1)*(self.box.height-2)])
+
+    def detail(self, poke):
+        global ev
+        ev = ""
+
+        poke = Poke(poke, 0)
+        desc_text = liner(poke.desc.text.replace("\n", " "), 29)
+        self.detail_box.resize(9+len(desc_text.split("\n")), 35)
+        self.detail_box.name_label.rechar(poke.name)
+        self.detail_box.add_ob(poke.ico, 3, 2)
+        self.detail_desc.rechar(desc_text)
+        self.detail_info.rechar("Type: ")
+        self.detail_info += se.Text(poke.type.name.capitalize(),
+                                    esccode=poke.type.color) + se.Text((f"""
+HP: {poke.hp}
+Attack: {poke.atc}
+Defense: {poke.defense}
+Initiative: {poke.initiative}"""))
+        with self.detail_box.center_add(self.map):
+            while True:
+                if ev in ["'e'", "Key.esc", "'q'"]:
+                    ev = ""
+                    break
+                std_loop()
+                time.sleep(0.05)
+                self.map.show()
+        self.detail_box.rem_ob(poke.ico)
+
+
+    def __call__(self, pokes):
+        global ev
+        ev = ""
+        self.idx = 0
+
+        p_dict = {i[1]: i[-1] for i in
+            sorted([(pokes[j]["type"], j, pokes[j]) for j in list(pokes)[1:]])}
+        self.obs = [se.Text(f"{i+1} {p_dict[poke]['name'] if poke in caught_poketes else '???'}", state="float")
+                for i, poke in enumerate(p_dict)]
+        self.add_c_obs()
+        with self.box.add(self.map, self.map.width-self.box.width, 0):
+            while True:
+                for event, idx, n_idx, add, idx_2 in zip(["'s'", "'w'"],
+                        [len(self.box.c_obs)-1, 0], [0, self.box.height-3],
```

Comment:
Is there any way for `self.box.c_obs` to be `0`?

If so, would a negative number break this?

Reply:
`self.box.c_obs` is the list of all Text labels displayed in the `ChooseBox` `self.box`. So it can't be negative.

Reply:
Ok. Thanks for clarification!